api/src/main/resources/marquez/db/migration/V47__dataset_symlinks.sql

```diff
@@ -0,0 +1,24 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+
+CREATE TABLE dataset_symlinks (
+  dataset_uuid      UUID,
+  name              VARCHAR(255) NOT NULL,
+  namespace_uuid    UUID REFERENCES namespaces(uuid),
+  type              VARCHAR(64),
+  is_primary        BOOLEAN DEFAULT FALSE,
+  created_at        TIMESTAMP NOT NULL,
+  updated_at        TIMESTAMP NOT NULL,
+  UNIQUE (namespace_uuid, name)
+);
+
+CREATE INDEX dataset_symlinks_dataset_uuid on dataset_symlinks (dataset_uuid);
+
+INSERT INTO dataset_symlinks (dataset_uuid, name, namespace_uuid, is_primary, created_at, updated_at)
+SELECT d.uuid, d.name, d.namespace_uuid, TRUE, d.created_at, d.updated_at FROM datasets d;
+
+DROP TYPE IF EXISTS DATASET_NAME;
+CREATE TYPE DATASET_NAME AS (
```

Comment:
Why do we need it?

Reply:
because of this:
```
WHERE CAST((:namespaceName, :datasetName) AS DATASET_NAME) = ANY(d.dataset_symlinks)
```
https://github.com/MarquezProject/marquez/pull/2087/files#diff-c687ca258e023847591c9b8c044adddaae1e7c1f6f9483928bf2871050cf6a63R77

I thought it's necessary in postgresql to check if pair of varchars is contained in array of pairs of varchars.

Reply:
Ok, so you can do `WHERE namespace_name IN X AND name in Y` because it can match namespace, and match name, but those belong to different symlinks, right?

Reply:
> Ok, so you can do `WHERE namespace_name IN X AND name in Y` because it can match namespace, and match name, but those belong to different symlinks, right?

Exactly, @mobuchowski